### ISSUE TITLE
Question and possibly related to Mod Code efficiency. And Is mod Smooth_Salad-IVYL a must? 

### ISSUE THREAD

**Issue by KaitoXion** [Priscillalala/FreeSkillSaturday#3]
Is Smooth_Salad-IVYL  a must include for using this mod?

Reason is cause I realize that every time when there is a mod updates even if its not related to this or your current mod, it will cause loading bug where the full function of that mod / mods cannot be loaded when the game 1st start up.
I had to restart the game again to get it to work properly. (So far as I observed.)

I think is due to something like redundancy as other mod directly access r2api modules instead of going through another.
I'll do some more further testing, and see if there are further problems before deciding if is best to move on.

Thank you for your creation, is very much appreciated.

**Comment by Priscillalala** [comment]
Yes, this mod depends on IVYL. It should not affect installation at all though

**Comment by KaitoXion** [comment]
Noted on your kind comment in regards to required IVYL mod dependency.

While not being sure what you meant about 
> should not affect installation at all though

well, so far after further running to this similar issue
> every time when there is a mod updates even if its not related to this or your current mod, it will cause loading bug where the full function of that mod / mods cannot be loaded when the game 1st start up.

The work around seems to be 1st complete booth (where there's mods changes / updates) then shut down, then reboot for proper game play.

Anyway, thanks for your creation.

**Comment by KaitoXion** [comment]
> Yes, this mod depends on IVYL. It should not affect installation at all though

Hi, this is just an additional information if you happen to ever be interested in the future. 

 I notice that the best way to test it is via **400 ticket, Mod: Hex3Mod**.
Some of the other few mod items do effected.

I notice that behavior changing things tend to get effected. Such are like item that can be use seems to be most obviously effected (??). **400 ticket, from Mod: Hex3Mod** appear to be most obvious one and easy to spot.

Reproduce method:
1. Change or update a mod. Then during 1st booth up of the  game. (2nd booth entirely new booth up of the client or the game will stop the issue from happening)
2. Obtain a 400 ticket.
3. Get another item from a chess to used up the ticket. (The ticket is suppose to be destroyed / used after being use)

Expected result:
> 400 ticket are suppose to be destroyed / used after opening a chest for item. The 400 ticket will turn grey like how potion does.

During Bug:
> 400 ticket never get destroy and can be repeatedly use.

Related Mods in current discussion:
>Hex3Mod
FreeSkillSaturday
IVYL

Sorry for the trouble. I'll update the main post as well.
I wasn't able to get a picture of it yet for now since am in a rush.
